Client/Application/Delegates/AppDelegate.swift

```diff
@@ -344,6 +305,58 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
       
         return shouldPerformAdditionalDelegateHandling
     }
+    
+    func handleReferralLookup(_ urp: UserReferralProgram, checkClipboard: Bool) {
+        let initialOnboarding =
+            Preferences.General.basicOnboardingProgress.value == OnboardingProgress.none.rawValue
+        
+        if initialOnboarding && UIPasteboard.general.hasStrings {
+            log.debug("Skipping URP call at app launch, this is handled in privacy consent onboarding screen")
+            return
+        }
+        
+        if Preferences.URP.referralLookupOutstanding.value == true {
+            var refCode: String?
+            
+            let savedRefCode: String? =
+                (!checkClipboard && ProcessInfo().operatingSystemVersion.majorVersion > 13) ?
```

Comment:
Shouldn't this drop the `majorVersion` check now that we show privacy consent screen on all versions?